google/cloud/bigquery/client.py

```diff
@@ -3190,6 +3190,7 @@ def query(
                 If ``job_config`` is not an instance of :class:`~google.cloud.bigquery.job.QueryJobConfig`
                 class.
         """
+        job_id_given = job_id is not None
```

Comment:
I'm not sure why it matters whether the user has passed in the `job_id`:  if we call `QueryJob._begin`, and the back-end returns `409 Conflict`, ISTM we should handle it the same either way.

Reply:
In Java the thought was if someone is providing their own job IDs and they get their job ID generation wrong, this could end up returning results for the wrong query.

Reply:
OK.